include/exadg/incompressible_navier_stokes/spatial_discretization/turbulence_model.cpp

```diff
@@ -325,14 +429,13 @@ TurbulenceModel<dim, Number>::vreman_model(scalar const & filter_width,
 
 template<int dim, typename Number>
 void
-TurbulenceModel<dim, Number>::wale_model(scalar const & filter_width,
-                                         tensor const & velocity_gradient,
-                                         double const & C,
-                                         scalar &       viscosity) const
+ViscosityModel<dim, Number>::wale_turbulence_model(scalar const & filter_width,
+                                                   tensor const & velocity_gradient,
+                                                   scalar const & shear_rate_squared,
+                                                   double const & C,
+                                                   scalar &       viscosity) const
 {
-  tensor S =
-    dealii::make_vectorized_array<Number>(0.5) * (velocity_gradient + transpose(velocity_gradient));
-  scalar S_norm_square = scalar_product(S, S);
+  scalar S_norm_square = dealii::make_vectorized_array<Number>(0.5) * shear_rate_squared;
```

Comment:
If I remember correctly, we tried to follow the notation according to the respective publications on the turbulence models. I am not sure if the implementation improves if we change all this here.